### ISSUE TITLE
Query Parameters are Not Fully Sanitized

### ISSUE THREAD

**Issue by gurumeditator** [atomicbits/scraml#59]
When generating client code with parameters that contain []:
```
...
queryParameters:

  param_name[]: 
...
```

Produces errors in the generated java resource:

```
...
 public CompletableFuture<Response<Measurement>> get(List<String> param_name[]
...

**Comment by rigolepe** [comment]
I'll release 0.4.11 later today with a fix for this one.

**Comment by gurumeditator** [comment]
I am using version 0.4.15 and noticed parameters are not getting sanitized:

**RAML**
```
queryParameters:

  filter_name[]: 
  ...
```

**Generated Resource**
```
       public CompletableFuture<Response<ServergroupInlineCollection>> get(Long count, List<String> filter_andOr[], List<String> filter_name[], List<String> filter_op[], List<String> filter_values[], List<String> scope_name[], List<String> scope_type[], String sort, Long start, String traceOn) {

         
           Map<String, HttpParam> params = new HashMap<String, HttpParam>();
           params.put("count", new SingleHttpParam(count));
params.put("scope_name[]", new RepeatedHttpParam(scope_name[]));
params.put("traceOn", new SingleHttpParam(traceOn));
params.put("filter_andOr[]", new RepeatedHttpParam(filter_andOr[]));
params.put("filter_name[]", new RepeatedHttpParam(filter_name[]));
params.put("filter_op[]", new RepeatedHttpParam(filter_op[]));
params.put("filter_values[]", new RepeatedHttpParam(filter_values[]));
params.put("scope_type[]", new RepeatedHttpParam(scope_type[]));
params.put("sort", new SingleHttpParam(sort));
params.put("start", new SingleHttpParam(start));
```

**Comment by rigolepe** [comment]
So, it includes the brackets in the query parameters. I'll see if there are other issues with parameter sanitization and fix it in a bugfix release this week.

**Comment by rigolepe** [comment]
Fixed in 0.4.16.

**Comment by gurumeditator** [comment]
Thanks Peter! I should be able to test on Monday.
